rust/lakesoul-datafusion/src/datasource/file_format/metadata_format.rs

```diff
@@ -573,3 +555,98 @@ fn make_sink_schema() -> SchemaRef {
         Field::new("msg", DataType::Utf8, false),
     ]))
 }
+
+
+/// File group to splits then flatten
+/// maintain the relational structure
+/// split -> (Vec<ParquetExec>, Vec<NullableCols>)
+async fn create_split_map(
+    conf: &FileScanConfig,
+    format: Arc<ParquetFormat>,
+    state: &SessionState,
+    pks: &[String],
+    target_schema: SchemaRef,
+    predicate: Option<Arc<dyn PhysicalExpr>>,
+) -> Result<HashMap<SplitDesc, (Vec<Arc<dyn ExecutionPlan>>, HashSet<String>)>> {
```

Comment:
Why using a HashMap as result collector

Reply:
change to tuple (SplitDesc, (Vec<_>, HashSet)) ?